NumLib/ODESolver/NonlinearSolver.cpp

```diff
@@ -45,6 +45,7 @@ bool NonlinearSolver<NonlinearSolverTag::Picard>::solve(
 
     bool error_norms_met = false;
 
+    MathLib::LinAlg::finalizeAssembly(x);
```

Comment:
is here really the right place to finalize `x`? The vector is given as a parameter to this function.